planetiler-core/src/main/java/com/onthegomap/planetiler/collection/FeatureGroup.java

```diff
@@ -60,27 +61,55 @@ public final class FeatureGroup implements Iterable<FeatureGroup.TileFeatures>,
   private final Stats stats;
   private final LayerStats layerStats = new LayerStats();
   private volatile boolean prepared = false;
+  private final TileOrder tileOrder;
 
-  FeatureGroup(FeatureSort sorter, Profile profile, Stats stats) {
+  public enum TileOrder {
+    TMS(0),
+    HILBERT(1);
+
+    private final int number;
+
+    private TileOrder(int number) {
+      this.number = number;
+    }
+  }
```

Comment:
You don't need to include the number in TileOrder definition - that gets added automatically and exposed as `TileOrder.TMS.ordinal()`.

One useful thing you can do with enums that I learned from [effective java](https://www.amazon.com/Effective-Java-Joshua-Bloch/dp/0134685997) is you can include a method definition in each enum value, so you could do something like:

```java
public enum TileOrder {
  TMS(TileCoord::encoded, TileCoord::decode),
  HILBERT(TileCoord::hilbertEncoded, TileCoord::hilbertDecode);

  private final ToIntFunction<TileCoord> encode;
  private final IntFunction<TileCoord> decode;

  private TileOrder(ToIntFunction<TileCoord> encode, IntFunction<TileCoord> decode) {
    this.encode = encode;
    this.decode = decode;
  }

  public int encode(TileCoord coord) {
    return encode.applyAsInt(coord);
  }

  public TileCoord decode(int encoded) {
    return decode.apply(encoded);
  }
}
```

Then you can use it later, so the details stay hidden from the consumer:

```java
int encoded = tileOrder.encode(feature.tile());
```